client/components/data/store-connection/index.jsx

```diff
@@ -70,9 +70,9 @@ class StoreConnection extends React.Component {
 			return React.createElement( this.props.component, this.state );
 		}
 
-		const child = React.Children.only( this.props.children );
-
-		return React.cloneElement( child, this.state );
+		return React.Children.map( this.props.children, child => {
```

Comment:
By way of explaining, we're removing the restriction of having only one child, and instead iterating through all children because moving the cart to the right makes `<CheckoutData>` [to have two children](https://github.com/Automattic/wp-calypso/pull/32832/files#diff-7f164a3b7deeca0666c8602e2b20e4acR49).